pinot-core/src/main/java/org/apache/pinot/core/plan/maker/InstancePlanMakerImplV2.java

```diff
@@ -61,7 +61,11 @@
   public static final String MAX_INITIAL_RESULT_HOLDER_CAPACITY_KEY = "max.init.group.holder.capacity";
   public static final int DEFAULT_MAX_INITIAL_RESULT_HOLDER_CAPACITY = 10_000;
   public static final String NUM_GROUPS_LIMIT = "num.groups.limit";
+  public static final String SEGMENT_TRIM_SIZE = "num.segment.trim.limit";
```

Comment:
Let's not make this configurable in step 1. Use `max(limit * 5, 5000)` (same as inter segment trim). You can use `GroupByUtils` to get the trim size from the `limit`

Reply:
Not making it configurable will change the default behavior?

Reply:
We want to configure whether to enable it, but not the threshold in step 1. If it is off, then it is the same as the current behavior